Tests/Tracking/CustomerIOImplementationTest.swift

```diff
@@ -129,6 +129,32 @@ class CustomerIOImplementationTest: UnitTest {
         XCTAssertEqual(profileIdentifyHookMock.profileIdentifiedCallsCount, 1)
     }
 
+    func test_identify_givenEmptyIdentifier_givenNoProfilePreviouslyIdentified_expectDoNotRunHooks() {
+        let givenIdentifier = ""
+        profileStoreMock.identifier = nil
+
+        customerIO.identify(identifier: givenIdentifier)
+
+        XCTAssertFalse(hooksMock.mockCalled)
+        XCTAssertNil(profileStoreMock.identifier)
+    }
+
+    func test_identify_givenEmptyIdentifier_givenProfileAlreadyIdentified_expectDoNotRunHooks_expectDoNotDeleteDeviceToken() {
+        let givenIdentifier = ""
+        let givenPreviouslyIdentifiedProfile = String.random
+        profileStoreMock.identifier = givenPreviouslyIdentifiedProfile
+
+        backgroundQueueMock.addTaskReturnValue = (
+            success: true,
+            queueStatus: QueueStatus.successAddingSingleTask
+        )
+
+        customerIO.identify(identifier: givenIdentifier)
+
+        XCTAssertFalse(hooksMock.mockCalled)
+        XCTAssertTrue(backgroundQueueMock.deviceTokensDeleted.isEmpty)
+    }
```

Comment:
```suggestion
```

Test not needed because empty identifier check not needed in `registerDeviceToken()`.

Reply:
Added a note [here](https://github.com/customerio/customerio-ios/pull/353#discussion_r1263864194) to support this.